go/parquet/encryption_read_config_test.go

```diff
@@ -0,0 +1,443 @@
+
+		int96md, _ := rgMeta.ColumnChunk(3)
+		// Read all rows in column
+		i = 0
+		for int96reader.HasNext() {
+			var (
+				val [1]parquet.Int96
+			)
+
+			// read one value at a time. the number of rows read is returned. values
+			// read contains the number of non-null rows
+			rowsRead, valuesRead, _ = int96reader.ReadBatch(1, val[:], nil, nil)
+			// ensure only 1 value is read
+			d.EqualValues(1, rowsRead)
+			// there are no null values
+			d.EqualValues(1, valuesRead)
+			// verify the value
+			var expectedValue parquet.Int96
+			binary.LittleEndian.PutUint32(expectedValue[:4], uint32(i))
+			binary.LittleEndian.PutUint32(expectedValue[4:], uint32(i+1))
+			binary.LittleEndian.PutUint32(expectedValue[8:], uint32(i+2))
+			d.Equal(expectedValue, val[0])
+			i++
+		}
+		d.EqualValues(i, int96md.NumValues())
+
+		if decryptConfigNum != 3 {
+			// Get column reader for the float column
+			colReader = rowGroupReader.Column(4)
+			floatReader := colReader.(*file.Float32ColumnChunkReader)
+
+			floatmd, _ := rgMeta.ColumnChunk(4)
+
+			i = 0
+			for floatReader.HasNext() {
+				var value [1]float32
+				// read one value at a time. the number of rows read is returned. values
+				// read contains the number of non-null rows
+				rowsRead, valuesRead, _ = floatReader.ReadBatch(1, value[:], nil, nil)
+				// ensure only 1 value is read
+				d.EqualValues(1, rowsRead)
+				// there are no null values
+				d.EqualValues(1, valuesRead)
+				// verify the value
+				expectedValue := float32(i) * 1.1
+				d.Equal(expectedValue, value[0])
+				i++
+			}
+			d.EqualValues(i, floatmd.NumValues())
+
+			// Get column reader for the double column
+			colReader = rowGroupReader.Column(5)
+			dblReader := colReader.(*file.Float64ColumnChunkReader)
+
+			dblmd, _ := rgMeta.ColumnChunk(5)
+
+			i = 0
+			for dblReader.HasNext() {
+				var value [1]float64
+				// read one value at a time. the number of rows read is returned. values
+				// read contains the number of non-null rows
+				rowsRead, valuesRead, _ = dblReader.ReadBatch(1, value[:], nil, nil)
+				// ensure only 1 value is read
+				d.EqualValues(1, rowsRead)
+				// there are no null values
+				d.EqualValues(1, valuesRead)
+				// verify the value
+				expectedValue := float64(i) * 1.1111111
+				d.Equal(expectedValue, value[0])
+				i++
+			}
+			d.EqualValues(i, dblmd.NumValues())
+		}
+
+		colReader = rowGroupReader.Column(6)
+		bareader := colReader.(*file.ByteArrayColumnChunkReader)
+
+		bamd, _ := rgMeta.ColumnChunk(6)
+
+		i = 0
+		for bareader.HasNext() {
+			var value [1]parquet.ByteArray
+			var def [1]int16
+
+			rowsRead, valuesRead, _ := bareader.ReadBatch(1, value[:], def[:], nil)
+			d.EqualValues(1, rowsRead)
+			expected := [10]byte{'p', 'a', 'r', 'q', 'u', 'e', 't', 0, 0, 0}
+			expected[7] = byte('0') + byte(i/100)
+			expected[8] = byte('0') + byte(i/10)%10
+			expected[9] = byte('0') + byte(i%10)
+			if i%2 == 0 {
+				d.Equal(1, valuesRead)
```

Comment:
what is this verifying exactly?  Should this be augmented with a test verifying errors/inequality when reading data from an encrypted file without keys?

Reply:
The lines here: https://github.com/apache/arrow/pull/11538/files/61ce47474d7ec42a80f1d6b354fbc28f4914edab#diff-cef277f99a3822e591750fc26b274b33e7be004a21833b2f38f5c438c1069178R382

and here: https://github.com/apache/arrow/pull/11538/files/61ce47474d7ec42a80f1d6b354fbc28f4914edab#diff-cef277f99a3822e591750fc26b274b33e7be004a21833b2f38f5c438c1069178R391

Verify that a panic happens when attempting to read from an encrypted file without proper keys.